src/Castle.Windsor.Extensions.DependencyInjection/WindsorScopeFactory.cs

```diff
@@ -16,29 +16,37 @@
 namespace Castle.Windsor.Extensions.DependencyInjection
 {
 	using System;
-	
+
 	using Castle.Windsor;
 	using Castle.Windsor.Extensions.DependencyInjection.Scope;
-	
+
 	using Microsoft.Extensions.DependencyInjection;
 
 	internal class WindsorScopeFactory : IServiceScopeFactory
 	{
 		private readonly IWindsorContainer scopeFactoryContainer;
+		private readonly ExtensionContainerRootScope rootScope;
 
-		public WindsorScopeFactory(IWindsorContainer container)
+		public WindsorScopeFactory(
+			IWindsorContainer container,
+			ExtensionContainerRootScope rootScope)
 		{
 			scopeFactoryContainer = container;
+			this.rootScope = rootScope;
 		}
 
 		public IServiceScope CreateScope()
 		{
-			var scope = ExtensionContainerScope.BeginScope(ExtensionContainerScope.Current);
-			
+			var scope =
+				ExtensionContainerScope
+					.BeginScope(
+						ExtensionContainerScope.Current,
+						rootScope);
```

Comment:
Any reason this need 5 lines rather than just 1, it isn't very long?

```csharp
var scope = ExtensionContainerScope.BeginScope(ExtensionContainerScope.Current, rootScope);
```

Reply:
fixed